tests/integration/acp/index/query_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package test_acp_index
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+	acpUtils "github.com/sourcenetwork/defradb/tests/integration/acp"
+)
+
+func TestACPWithIndex_UponQueryingPrivateDocWithoutIdentity_ShouldNotFetch(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Test acp, querying private doc without identity should not fetch",
+		Actions: []any{
+			testUtils.AddPolicy{
+				Identity:         acpUtils.Actor1Identity,
+				Policy:           userPolicy,
+				ExpectedPolicyID: "53980e762616fcffbe76307995895e862f87ef3f21d509325d1dc772a770b001",
```

Comment:
suggestion: It would be nice to add the policy ID above the policy definition as a comment for future reference. Like if someone what's to use it elsewhere it would be easy for them to quickly grab it.